integration_tests/gdk/components/test_integ_PublishCommand.py

```diff
@@ -119,6 +119,108 @@ def test_publish_run_with_bucket_argument(mocker, get_service_clients, mock_proj
     assert spy_get_caller_identity.call_count == 1  # Get account number
 
 
+def test_publish_run_with_region_and_options_argument(mocker, get_service_clients, mock_project_config):
+    mock_build_dir_exists = mocker.patch(
+        "gdk.common.utils.dir_exists",
+        return_value=True,
+    )
+    pc = mock_project_config.return_value
+    mock_iter_dir = mocker.patch("pathlib.Path.iterdir", return_value=[Path("hello_world.py")])
+
+    mock_glob = mocker.patch("pathlib.Path.glob", return_value=[Path("hello_world.py")])
+    file_name = (
+        Path(pc["gg_build_recipes_dir"]).joinpath("{}-{}.json".format(pc["component_name"], pc["component_version"])).resolve()
+    )
+
+    spy_get_caller_identity = mocker.spy(get_service_clients["sts_client"], "get_caller_identity")
```

Comment:
We should start using TestDoubles/Fakes instead of doing all this mocking. This test is not giving us confidence thing will behave correctly. Asserting agains call_count won't tell us it did what we expected it to do.

I am ok with this as long as we follow up on a PR improving this tests

Reply:
I changed mock to spy wherever possible.  This is an integ test and I'm mocking other library methods/objects. L-151 to L-160  show what args are used.